components/ws-manager-api/core.proto

```diff
@@ -318,6 +318,8 @@ message IDEImage {
     string desktop_ref = 2;
     // supervisor_ref is a reference to an OCI image used as supervisor
     string supervisor_ref = 3;
+    // desktop_plugin_ref is an optional reference to an OCI image used for serving desktop IDE plugin
+    string desktop_plugin_ref = 4;
```

Comment:
@csweichel @aledbf 

Maybe it is better if we switch to just `refs` to make it easier to change IDE layers in backward compatible way without relying on deployments of ws-manager.

Reply:
How often do we expect these layers to change down the road?

Reply:
We maybe would like to make some changes for VS Code optimisation too.

The thing is that I'm not sure why it should be specific here, ws-manager does not seem to care about the semantic.  Semantic is defined by the IDE service. ws-manager just need to apply them. Maybe we should rather then remove this dependency.

Reply:
We decided first to deal with extracting IDE service from Server.